File: backend/app/schemas/review.py
"""
Schemas de reseña.
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class ReviewCreate(BaseModel):
    order_id: int
    rating: float = Field(ge=1.0, le=5.0)
    comment: Optional[str] = None


class ReviewResponse(BaseModel):
    id: int
    order_id: int
    customer_id: int
    driver_id: Optional[int] = None
    rating: float
    comment: Optional[str] = None
    created_at: datetime
    customer_name: Optional[str] = None

    class Config:
        from_attributes = True

File: mobile/lib/models/review.dart
import '../core/format.dart';

class Review {
  Review({
    required this.id,
    required this.orderId,
    required this.customerId,
    this.driverId,
    required this.rating,
    this.comment,
    this.createdAt,
    this.customerName,
  });

  final int id;
  final int orderId;
  final int customerId;
  final int? driverId;
  final double rating;
  final String? comment;
  final DateTime? createdAt;
  final String? customerName;

  factory Review.fromJson(Map<String, dynamic> j) => Review(
        id: (j['id'] as num).toInt(),
        orderId: (j['order_id'] as num?)?.toInt() ?? 0,
        customerId: (j['customer_id'] as num?)?.toInt() ?? 0,
        driverId: (j['driver_id'] as num?)?.toInt(),
        rating: (j['rating'] as num?)?.toDouble() ?? 0,
        comment: j['comment'] as String?,
        createdAt: Fmt.parseUtc(j['created_at'] as String?),
        customerName: j['customer_name'] as String?,
      );
}
